deps/spidershim/src/v8string.cc

```diff
@@ -96,27 +95,40 @@ MaybeLocal<String> String::NewFromUtf8(Isolate* isolate, const char* data,
   if (!twoByteChars) {
     return MaybeLocal<String>();
   }
-  JS::RootedString str(cx, JS_NewUCString(cx, twoByteChars.release(), twoByteLen));
+
+  JSString* str =
+    type == v8::NewStringType::kInternalized ?
+      JS_AtomizeAndPinUCString(cx, reinterpret_cast<const char16_t*>(twoByteChars.release())) :
+      JS_NewUCString(cx, twoByteChars.release(), twoByteLen);
+
   if (!str) {
     return MaybeLocal<String>();
   }
+
+  JS::RootedString rootedStr(cx, str);
```

Comment:
> Why do you need to root the string here? The internal::Local<String>::New() call below does that for you, and nothing can GC here between these two points.

This is the existing behavior, as you originally implemented it, and I'm not changing it in this PR.